src/mlpack/methods/ann/activation_functions/softplus_function.hpp

```diff
@@ -51,9 +51,21 @@ class SoftplusFunction
    */
   static double Fn(const double x)
   {
+    const double val = std::log(1 + std::exp(x));
     if (x < DBL_MAX)
```

Comment:
Why this check? If you're checking the value of the outputs, this becomes redundant.

Reply:
Let me change it and see if any test fails..